tools/lang_demo.py

```
@@ -0,0 +1,388 @@
+
+
+def _pick_default_branch(expr: str) -> str:
+    """Resolve keycode_helper.c's conditional returns for the *resting* keycap:
+    state_flags = 0 (so MORE_TEXT / MODS_AS_TEXT off → the icon branch, not text),
+    num_lock / caps_lock off. Picks the branch a freshly-booted key would draw."""
+    t = _split_ternary(expr)
+    if t is None:
+        return expr.strip()
+    cond, a, b = t
+    # evaluate the condition under the resting state
+    if '!= 0' in cond:        val = False     # (state_flags & X) != 0  -> 0 != 0 -> false
```

Ruff (0.15.18)

[error] Multiple statements on one line (colon)

(E701)